log/logger.go

```diff
@@ -1,27 +1,106 @@
 package log
 
 import (
+	"context"
+	"fmt"
 	"io"
-	"log"
+	"log/slog"
+	"maps"
 	"os"
 	"path/filepath"
+	"slices"
+	"strings"
+	"time"
 
 	pkgerrors "github.com/pkg/errors"
 	"github.com/zimmski/osutil/bytesutil"
 )
 
+// AttributeKey defines a key for attributes handed to the structural logger.
+type AttributeKey string
+
+const (
+	AttributeKeyLanguage   AttributeKey = "Language"
+	AttributeKeyModel                   = "Model"
+	AttributeKeyRepository              = "Repository"
+	AttributeKeyResultPath              = "ResultPath"
+	AttributeKeyTask                    = "Task"
+)
+
+type Flags int
+
+const (
+	FlagMessageOnly = 0
+	FlagDate        = 1 << iota
+	FlagTime
+	FlagStandard = FlagDate | FlagTime
+)
```

Comment:
Docstrings. Not crazy necessary for the keys/flags themselves but at least for the flag.

Reply:
`// Flags defines how log messages should be printed.`